### PR TITLE
CPV: add LM-L0 delay subtraction to RawToDigitConverterSpec

### PR DESCRIPTION
**Pull request by sevdokim**
Hello! LM-L0 delay subtraction is added to RawToDigitConverterSpec. Also it is taken into account in simulation/RawWriter.cxx.
Also ccdb timestamp is fetched  from hbfutils (was 'now' previously) in there.

### CODE REVIEW

**Comment by sevdokim** [comment]
hello @shahor02 
can you please have a look at the ci errors? they look unrelated to cpv code. should I worry about them?

File: Detectors/CPV/workflow/src/RawToDigitConverterSpec.cxx [hunk 1]
```diff
@@ -58,8 +59,47 @@ void RawToDigitConverterSpec::init(framework::InitContext& ctx)
...
+  if (!updateOnlyOnce) {
+    ctx.inputs().get<o2::ctp::TriggerOffsetsParam*>("trigoffset");
+
+    std::decay_t<decltype(ctx.inputs().get<o2::cpv::Pedestals*>("peds"))> pedPtr{};
+    std::decay_t<decltype(ctx.inputs().get<o2::cpv::BadChannelMap*>("badmap"))> badMapPtr{};
+    std::decay_t<decltype(ctx.inputs().get<o2::cpv::CalibParams*>("gains"))> gainsPtr{};
+
+    if (!mIsPedestalData) {
+      pedPtr = ctx.inputs().get<o2::cpv::Pedestals*>("peds");
+      mPedestals = pedPtr.get();
```
Comment (shahor02) [inline Detectors/CPV/workflow/src/RawToDigitConverterSpec.cxx]: same as before: why don't you simply use 
```
auto pedPtr = ctx.inputs().get<o2::cpv::Pedestals*>("peds");
mPedestals = pedPtr.get();
```
?

File: Detectors/CPV/simulation/src/RawWriter.cxx [hunk 2]
```diff
@@ -40,7 +42,11 @@ void RawWriter::init()
     mRawWriter->registerLink(link.feeId, link.cruId, link.linkId, link.endPointId, rawFileName.data());
   }
 
+  // Lm-L0 delay
+  mLM_L0_delay = o2::ctp::TriggerOffsetsParam::Instance().LM_L0;
```
Comment (shahor02) [inline Detectors/CPV/simulation/src/RawWriter.cxx]: I don't see where you fetch the TriggerOffsetsParam from the CCDB, seem you are always using default in digi2raw while for decoding you use it from CCDB?

Comment (sevdokim) [reply Detectors/CPV/simulation/src/RawWriter.cxx]: Yes, it is correct. If you suggest how to fetch it using BasicCCDBManager then I'll insert it.

File: Detectors/CPV/workflow/src/RawToDigitConverterSpec.cxx (hunk 1 above)
Comment (sevdokim) [reply Detectors/CPV/workflow/src/RawToDigitConverterSpec.cxx]: There was a reason for that when code was organized in different manner. I'll try to change it now.

File: Detectors/CPV/simulation/src/RawWriter.cxx (hunk 2 above)
Comment (shahor02) [reply Detectors/CPV/simulation/src/RawWriter.cxx]: Exactly like you do for `CPV/Calib/Gains`, using the timestamp from the hbfutils. Simply, for the ConfigurableParams you don't need to assign the result of `ccdbMgr.get<o2::ctp::TriggerOffsetsParam>("CTP/Config/TriggerOffsets")`

Comment (sevdokim) [reply Detectors/CPV/simulation/src/RawWriter.cxx]: thanks, got it!

**Comment by sevdokim** [comment]
Please don't merge yet. I spotted an error.

**Comment by sevdokim** [comment]
hi! the bug is fixed. please merge if possible

**Comment by sevdokim** [comment]
hi @shahor02 
this PR has no updates for a while. can it be merged or we need to change something?

**Review by shahor02 — APPROVED** [review]

**Pull request merged**